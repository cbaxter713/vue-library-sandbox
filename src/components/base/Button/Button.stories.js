import Button from "./Button.vue";

// More on default export: https://storybook.js.org/docs/vue/writing-stories/introduction#default-export
export default {
  title: "Base/Button",
  component: Button,
  // More on argTypes: https://storybook.js.org/docs/vue/api/argtypes
  argTypes: {
    size: {
      control: { type: "select" },
      options: ["sm", "md", "lg"],
    },
    theme: {
      control: { type: "select" },
      options: ["primary", "secondary", "action"],
    },
  },
};

// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
const Template = (args) => ({
  // Components used in your story `template` are defined in the `components` object
  components: { Button },
  // The story's `args` need to be mapped into the template through the `setup()` method
  setup() {
    return { args };
  },
  // And then the `args` are bound to your component with `v-bind="args"`
  template: '<Button v-bind="args" />',
});

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args
Primary.args = {
  label: "Default",
};

export const Secondary = Template.bind({});
Secondary.args = {
  label: "Secondary",
  theme: "secondary",
};

export const Large = Template.bind({});
Large.args = {
  size: "lg",
  label: "Large",
};

export const Small = Template.bind({});
Small.args = {
  size: "sm",
  label: "Small",
};
