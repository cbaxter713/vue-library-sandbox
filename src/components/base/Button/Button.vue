<script setup lang="ts">
export interface ButtonProps {
  label: string;
  loading?: boolean;
  theme: "primary" | "secondary" | "action";
  size: "sm" | "md" | "lg";
}

const props = withDefaults(defineProps<ButtonProps>(), {
  size: "md",
  theme: "primary",
});

// const props = defineProps({
//   label: String,
//   /**
//    * Whether or not it is loading
//    */
//   loading: Boolean,
//   theme: {
//     type: String,
//     default: 'primary'
//   },
//   size: String
// });
</script>

<template>
  <button :class="[`btn btn-${theme} btn-size-${size}`, { loading: loading }]">
    {{ label }}
  </button>
</template>

<style scoped lang="scss">
.btn {
  appearance: none;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 3rem;
  padding: 0 2rem;
  border: 2px solid maroon;
  color: white;
  cursor: pointer;

  &-primary {
    background-color: $color-primary;
  }

  &-secondary {
    background-color: $color-secondary;
  }

  &-action {
    background-color: orange;
  }

  &-size-sm {
    height: 2rem;
  }

  &-size-lg {
    height: 4rem;
  }
}
</style>
