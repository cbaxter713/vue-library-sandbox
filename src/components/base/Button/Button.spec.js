import { mount } from "@vue/test-utils";
import Button from "./Button.vue";

const defaultLabel = "Button Test";
const defaultWrapper = mount(Button, { props: { label: defaultLabel } });

describe("Button", () => {
  it("Should match the default snapshot", () => {
    expect(defaultWrapper.element).toMatchSnapshot();
  });

  it("Should display the passed in label prop", () => {
    expect(defaultWrapper.find("button").text()).toEqual(defaultLabel);
  });
});
