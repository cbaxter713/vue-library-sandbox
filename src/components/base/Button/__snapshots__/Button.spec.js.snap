// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match the default snapshot 1`] = `
<button
  class="btn btn-primary btn-size-md"
>
  Button Test
</button>
`;
